x-pack/legacy/plugins/maps/public/connected_components/widget_overlay/layer_control/layer_toc/toc_entry/view.test.js

```diff
@@ -83,5 +80,33 @@ describe('TOCEntry', () => {
 
       expect(component).toMatchSnapshot();
     });
+
+    test('should shade background if selected layer is the same', async () => {
+      const component = shallowWithIntl(<TOCEntry {...defaultProps} selectedLayer={mockLayer} />);
+
+      // Ensure all promises resolve
+      await new Promise(resolve => process.nextTick(resolve));
+      // Ensure the state changes are reflected
+      component.update();
+
+      expect(component).toMatchSnapshot();
+    });
+
+    test('should not shade background if selected layer is not the same', async () => {
```

Comment:
This test case is not really needed and just bloats the snapshot. The other tests cases render without `mapTocEntry-isSelected` so I think that covers the negative case. Maybe selectedLayer={differentLayer} to default props.

Reply:
after offline discussion: it's needed to distinguish between the case when there's no selected a layer at all, and when there is one (but not _this_ one).